src/native/char-at.ts

```diff
@@ -1,15 +1,21 @@
 import type { Split } from './split.js'
+import type {
+  All,
+  IsStringLiteral,
+  IsNumberLiteral,
+} from '../internal/literals.js'
 
 /**
  * Gets the character at the given index.
  * T: The string to get the character from.
  * index: The index of the character.
  */
-export type CharAt<T extends string, index extends number> = string extends T
-  ? string
-  : number extends index
-  ? string
-  : Split<T>[index]
+export type CharAt<T extends string, index extends number> = All<
```

Comment:
I used `All` instead of nesting ternary operators wherever we need to check >=2 params 

I'm trying to avoid this:
```js
someCondition
 ? ...
   ? ...
     ? ...
       ? ...
       : ...
     : ...
   : ...
 : ...
```

Reply:
clever.. wouldn't `IsStringLiteral<A | B | C>` suffice?

Reply:
Kind of -- I think I can simplify a little bit.
Many of the multi-param functions need both `IsStringLiteral` and `IsNumberLiteral`, and those can't be combined to a union.

I also think this only works now that we added the unary tuple pattern -- I want to say I tried combining earlier, and something didn't work correctly

Reply:
*Updated*